jmclient/jmclient/wallet_rpc.py

```diff
@@ -375,21 +375,44 @@ def set_token(self, wallet_name):
         # startup, so any failure to exist here is a logic error:
         self.wss_factory.valid_token = self.cookie
 
-    def get_POST_body(self, request, keys):
+    def get_POST_body(self, request, required_keys, optional_keys=None):
         """ given a request object, retrieve values corresponding
-        to keys keys in a dict, assuming they were encoded using JSON.
-        If *any* of the keys are not present, return False, else
-        returns a dict of those key-value pairs.
+        to keys in a dict, assuming they were encoded using JSON.
+        If *any* of the required_keys are not present, return False, else
+        returns a dict of those key-value pairs and any of the optional key-value pairs.
         """
         assert isinstance(request.content, BytesIO)
         # we swallow any formatting failure here:
         try:
             json_data = json.loads(request.content.read().decode(
                 "utf-8"))
-            return {k: json_data[k] for k in keys}
+            required_body = {k: json_data[k] for k in required_keys}
+
+            if optional_keys is not None:
+                optional_body = {k: json_data[k] for k in optional_keys if k in json_data}
+                # required keys take precedence over optional keys
+                optional_body.update(required_body)
+                return optional_body
+
+            return required_body
         except:
             return False
 
+    def parse_tumbler_options_from_json(self, tumbler_options_json):
+        parsed_tumbler_options = {}
+
+        for key, val in tumbler_options_json.items():
+            if isinstance(val, list):
+                # Convert JSON lists to tuples.
+                parsed_tumbler_options[key] = tuple(val)
+            elif key == 'order_choose_fn':
+                # Todo: No support for custom order choose function via API yet.
```

Comment:
In case someone is curious (and for future reference) - these are the options, as of now: 

```python
order_choose_algorithms = {
    'random_under_max_order_choose': '-R',
    'cheapest_order_choose': '-C',
    'weighted_order_choose': '-W',
    'fidelity_bond_weighted_order_choose': '-F'
}
```

See `cli_options.py`

https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/50f5228498c670258740cd3cab485bcb39e177a3/jmclient/jmclient/cli_options.py#L16